src/routes/controllers/follow.controller.js

```diff
@@ -0,0 +1,65 @@
+import models from '../../db/models';
+import {
+  errorResponse,
+  successResponse,
+  getUserbyUsername,
+} from '../../utils/helpers.utils';
+
+const { User } = models;
+
+/** ,
+ * Follow A User
+ * @param {object} req
+ * @param {object} res
+ * @returns {object} followed user object
+ */
+const follow = async (req, res) => {
```

Comment:
I noticed you are handling both follow and unfollow using this controller, instead doing it this way, why not create two endpoint, one for `follow` and another for `unfollow`. It does not make sense checking the url to see if it includes `unfollow` to handle the logic for unfollowing a user. I feel this should separated, because right now this controller is doing too much.

Reply:
Okay I will implement this feedback now

Reply:
This has been resolved